src/main/java/org/jabref/gui/entryeditor/FxFileAnnotationTab.java

```diff
@@ -0,0 +1,223 @@
+package org.jabref.gui.entryeditor;
+
+import java.time.LocalDateTime;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import javafx.beans.property.SimpleStringProperty;
+import javafx.beans.property.StringProperty;
+import javafx.beans.value.ChangeListener;
+import javafx.beans.value.ObservableValue;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.scene.control.ComboBox;
+import javafx.scene.control.Label;
+import javafx.scene.control.ListView;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.SelectionMode;
+import javafx.scene.control.TextArea;
+import javafx.scene.control.Tooltip;
+import javafx.scene.layout.ColumnConstraints;
+import javafx.scene.layout.GridPane;
+import javafx.scene.layout.Priority;
+import javafx.scene.layout.Region;
+import javafx.scene.text.Text;
+
+import org.jabref.gui.BasePanel;
+import org.jabref.gui.IconTheme;
+import org.jabref.gui.JabRefFrame;
+import org.jabref.gui.fieldeditors.FieldEditorFX;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.pdf.FileAnnotationCache;
+import org.jabref.model.entry.FieldName;
+import org.jabref.model.pdf.FileAnnotation;
+
+import static org.jabref.model.pdf.FileAnnotationType.NONE;
+
+public class FxFileAnnotationTab extends EntryEditorTab {
```

Comment:
Why don't you use the Scence Builder to create your fxml?
http://gluonhq.com/open-source/scene-builder/